examples/hosts/iframe-host/src/inner.ts

```diff
@@ -24,26 +28,62 @@ async function getFluidObjectAndRender(container: Container, div: HTMLDivElement
     view.render(div, { display: "block" });
 }
 
-export async function runInner(divId: string) {
-    const div = document.getElementById(divId) as HTMLDivElement;
-
+async function loadContainer(
+    documentId: string,
+    createNew: boolean,
+): Promise<IContainerProxy> {
     const documentServiceFactory = await InnerDocumentServiceFactory.create();
+    const urlResolver = await InnerUrlResolver.create();
+
     const module = { fluidExport: TodoContainer };
     const codeLoader = { load: async () => module };
 
     const loader = new Loader({
-        urlResolver: documentServiceFactory.urlResolver,
+        urlResolver,
         documentServiceFactory,
         codeLoader,
     });
 
-    const url = documentServiceFactory.resolvedUrl.url;
-    const documentId = decodeURIComponent(url.replace("fluid://localhost:3000/tinylicious/", ""));
-    const container = await loader.resolve({ url: documentId });
+    let container: Container;
+
+    if (createNew) {
+        // We're not actually using the code proposal (our code loader always loads the same module regardless of the
+        // proposal), but the Container will only give us a NullRuntime if there's no proposal.  So we'll use a fake
+        // proposal.
+        container = await loader.createDetachedContainer({ package: "no-dynamic-package", config: {} });
```

Comment:
>container = await loader.createDetachedContainer({ package: "no-dynamic-package", config: {} }); [](start = 8, length = 96)

not necessary in this PR, but we should create a follow up issue to support proxying this in from the loader